### DISCUSSION TITLE
How to pass --preview argument to black

### DISCUSSION THREAD

**Discussion by simgunz** [akaihola/darker#350] [Q&A] [answered]
Hi,
in order to enable the experimental string processing of black, it is necessary to pass the `--preview` argument to black from the command line and apparently there is not a way to configure this behavior in black configuration file. How can I pass the `--preview` to black while using darker?

**Comment by akaihola** [comment, marked as answer]
**Update:** Design discussion (custom options vs. `--extra-black-options`) now in #589, and `--extra-black-options` implementation discussion in #351.

Arbitrary arguments to Black aren't yet supported.

Darker invokes Black directly through its internal Python API. The Black configuration is actually first read by Darker, and some options are passed on to Black. But `--preview` isn't currently one of those options.

We could either add `--preview` to the passed options, or add an `--extra-black-options="--preview ..."` kind of a command line argument to Darker.

What do you think of these options @simgunz?

**Reply by simgunz** [reply]
Thanks for the fast reply. I like it more the second one: `--extra-black-options="--preview ..."`

**Reply by akaihola** [reply]
@simgunz, this is now tracked in #351.

**Comment by akaihola** [comment]
#589 now proposes a `--preview` command line option.

I vote against adding corresponding configuration file options in the `[tool.darker]` section since the user can simply add
```toml
[tool.black]
preview = true
```
in the same file.

Discussion on the choice between `--preview` and something like `--extra-black-options="--preview"` is best continued in #589.
